### ISSUE TITLE
Retrieving request parameters returns null while testing

### ISSUE THREAD

**Issue by dealloc** [laravel/framework#13365]
When you run the application with PHPUnit and you test a JSON api the $request->get will always return null, but the $request->all() will return all parameters like normal. This causes it to pass form validation (which I assume uses $request->all()) but when accessing the parameters it'll return null (which is obviously a problem).

I demonstrated the problem as shown below:

using $this->post and adding the Accept header manually:
![1461877136](https://cloud.githubusercontent.com/assets/2164354/14901679/fffbafdc-0d96-11e6-9807-03c91a9885b8.png)

using $this->json and letting laravel take care of the headers:
![1461877113](https://cloud.githubusercontent.com/assets/2164354/14901678/ffd4d9b6-0d96-11e6-8c4c-7a473162b95e.png)

Output generated for both is done with a dd($this) in the CreateUserCommand which looks like this:
```php
class CreateUserCommand
{
	private $fname;
	private $lname;
	private $email;
	private $password;

	public function __construct(CreateUserRequest $request)
	{
		$this->fname = $request->get('fname');
		$this->lname = $request->get('lname');
		$this->email = $request->get('email');
		$this->password = $request->get('password');
		dd($this);
	}

        // getters for all properties go here
}
```

### TLDR
``` $this->json ``` will cause ``` $request->get ``` to return NULL at all times
``` $this->post ``` will cause the ``` $request->get ``` to return values as normal
big problem: due ``` $request->all ``` to pass, Form validation will pass and the app will crash later on

### EDIT
Found a possible duplicate of this issue #13101 but he doesn't mention form validation passing nor more specific what he did to reproduce steps

**Comment by James-ONeill** [comment]
I've been looking into this issue since earlier in the week and discovered that `$request->get()` doesn't seem to be in the Laravel docs or the API reference for `\Illuminate\Http\Request`. The method is inherited from `Symfony\Components\HttpFoundation\Request`.

`$request->input()` seems to do do the same thing and is working for me in tests using `$request->json()`. It also has the added benefit of supporting dot notation for array parameters.

**Comment by dealloc** [comment]
Now that you mention it, the ``` $request->get() ``` indeed doesn't appear in the framework documentation. You'd expect it to work though..
